### DISCUSSION TITLE
Why is Read10X not creating a Large List?

### DISCUSSION THREAD

**Discussion by anniepu** [satijalab/seurat#4585] [Q&A] [answered]
Hi everyone, 

I am trying to read Cellranger Count output files into R using Read10X(). I have a folder containing all 3 required files: barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz. 

On a previous (completely separate experiment) when I use Read10X() I get a Large List in the environment, but for this experiment I'm only getting a Large dgCMatrix. As a result I'm not able to use the downstream commands like CreateSeuratObject() and creating Gene Expression and Antibody Capture assays.

The FastQ files were processed through Cellranger the exact same way both times. Any insight would be appreciated!

**Comment by cswoboda** [comment, marked as answer]
Hey, not the seurat team, just a user looking to help out. Documentation on the command suggests this might be something you want to look into: "If features.csv indicates the data has multiple data types, a list containing a sparse matrix of the data from each type will be returned. Otherwise a sparse matrix containing the expression data will be returned. " 

You should be able to make seurat objects with large dgCMatrices though, im not sure what your barrier is there.

**Reply by anniepu** [reply]
Thanks! The problem is that I can't seem to distinguish my Gene Expression and Antibody Capture assays after creating Seurat object if I just proceed with the dgCMatrix, because it seems like it only contains Gene Expression data? I'm not sure what's happening here.

**Reply by cswoboda** [reply]
That could be the case. Something you could try doing is unzipping the features.csv from the gz format. If you believe that data should have been split between gene expression and antibody capture, maybe providing it in a zipped format is making it harder to distinguish (im sorry, i've never done the antibody capture before so someone else may need to provide advice on this).

**Reply by samuel-marsh** [reply]
Hi,  

So the `Read10X` function automatically parses the files output by Cell Ranger and if there are multiple modalities it returns a list with each entry being a sparse matrix of that modality.  If there is only one modality present it simply returns a sparse matrix.
See description in manual entry for `Read10X`.[https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L747-L749](https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/preprocessing.R#L747-L749)

I would go back and double check your original data and output from Cell Ranger as that seems to me to be the likely issue here.

Best,
Sam

**Reply by anniepu** [reply]
Thanks so much for your help Sam! I was able to rerun Cellranger and it has solved the problem.
